### DISCUSSION TITLE
matrix data input for finufft.nufft2d1()

### DISCUSSION THREAD

**Discussion by Bruce-Yu** [flatironinstitute/finufft#472] [FINUFFT in applications, mathematical definitions]
Hi, I have a matrix data input and the corresponding matrix for coordinates (from MRI k-space). Would it be possible to use   finufft.nufft2d1() to reconstruct the image with the two input matrices?
It seems finufft.Plan() is the solution as it accepts input data matrix. But plan.setpts(x, y) only accepts vector (not matrix) for coordinate.

Any suggestion/clue is highly appreciated.

**Comment by ahbarnett** [comment]
It sounds like you're using FINUFFT from Python. I can't help with a full reconstruction code (the other MRI users might help with that; see other discussions.) However, for simply performing a type-1 NUFFT (which if you had good gridding weights would actually be a reconstruction..), you need separate vecotrs for the x and y coords of the NU (k-space) pts. If your matrix were stored under the hood in the points-fast-dimensions-slow ordering, you could pass in views to the two such vectors. However, if it were the transpose, you'd have to copy out the x and y coords into separate vectors. Then follow `python/finufft/examples/simple2d1.py`
